holoviews/plotting/bokeh/util.py

```diff
@@ -148,27 +156,166 @@ def compute_plot_size(plot):
+                    "%s value was ignored because absolute width and "
+                    "height values were provided. Either supply "
+                    "explicit frame_width and frame_height to achieve "
+                    "desired aspect OR supply a combination of width "
+                    "or height and an aspect value." % aspect_type)
+        elif fixed_width and responsive:
+            height = None
+            responsive = False
+            if logger:
+                logger.warning("responsive mode could not be enabled "
+                               "because fixed width and aspect were "
+                               "specified.")
+        elif fixed_height and responsive:
+            width = None
+            responsive = False
+            if logger:
+                logger.warning("responsive mode could not be enabled "
+                               "because fixed height and aspect were "
+                               "specified.")
+        elif responsive == 'width':
+            sizing_mode = 'scale_width'
+        elif responsive == 'height':
+            sizing_mode = 'scale_height'
+
+
+    if responsive == 'width' and fixed_width:
+        responsive = False
+        if logger:
+            logger.warning("responsive width mode could not be enabled "
+                           "because a fixed width was defined.")
+    if responsive == 'height' and fixed_height:
+        responsive = False
+        if logger:
+            logger.warning("responsive height mode could not be enabled "
+                           "because a fixed height was defined.")
+
+    match_aspect = False
+    aspect_scale = 1
+    aspect_ratio = None
+    if (fixed_width and fixed_height):
+        pass
+    elif data_aspect or aspect == 'equal':
+        match_aspect = True
+        if fixed_width or not fixed_height:
+            height = None
+        if fixed_height or not fixed_width:
+            width = None
+        aspect_scale = 1 if aspect == 'equal' else data_aspect
+    elif isnumeric(aspect):
+        if responsive:
+            aspect_ratio = aspect
+        elif fixed_width:
+            frame_width = actual_width
+            frame_height = int(actual_width/aspect)
+            width, height = None, None
+        else:
+            frame_width = int(actual_height*aspect)
+            frame_height = actual_height
+            width, height = None, None
+    elif aspect is not None and logger:
+        logger.warning('aspect value of type %s not recognized, '
+                       'provide a numeric value, \'equal\' or '
+                       '\'square\'.')
+
+    return ({'aspect_ratio': aspect_ratio,
+             'aspect_scale': aspect_scale,
+             'match_aspect': match_aspect,
+             'sizing_mode' : sizing_mode},
+            {'frame_width' : frame_width,
+             'frame_height': frame_height,
+             'plot_height' : height,
+             'plot_width'  : width})
```

Comment:
It isn't clear to me from the docstring why two dictionaries of settings are being returned here...

Reply:
I'll add a docstring.